internal/exec/exec.go

```diff
@@ -18,36 +18,47 @@ import (
 	"github.com/neelance/graphql-go/trace"
 )
 
+type Logger interface {
+	Log(ctx context.Context, value interface{})
+}
+
+type DefaultLogger struct{}
+
+func (l *DefaultLogger) Log(_ context.Context, value interface{}) {
+	err := makePanicError(value)
+	const size = 64 << 10
+	buf := make([]byte, size)
+	buf = buf[:runtime.Stack(buf, false)]
+	log.Printf("%s\n%s", err, buf)
+}
+
 type Request struct {
 	selected.Request
 	Limiter chan struct{}
 	Tracer  trace.Tracer
+	Logger  Logger
 }
 
 type fieldResult struct {
 	name  string
 	value []byte
 }
 
-func (r *Request) handlePanic() {
-	if err := recover(); err != nil {
-		r.AddError(makePanicError(err))
+func (r *Request) handlePanic(ctx context.Context) {
+	if value := recover(); value != nil {
+		r.Logger.Log(ctx, value)
+		r.AddError(makePanicError(value))
 	}
 }
 
 func makePanicError(value interface{}) *errors.QueryError {
-	err := errors.Errorf("graphql: panic occurred: %v", value)
-	const size = 64 << 10
-	buf := make([]byte, size)
-	buf = buf[:runtime.Stack(buf, false)]
-	log.Printf("%s\n%s", err, buf)
-	return err
+	return errors.Errorf("graphql: panic occurred: %v", value)
```

Comment:
`makePanicError` is also called from `execFieldSelection` where the stack trace now gets lost.